engine/src/main/java/org/terasology/engine/core/GameThread.java

```diff
@@ -23,9 +29,22 @@
     private static volatile Thread gameThread;
     private static BlockingDeque<Runnable> pendingRunnables = Queues.newLinkedBlockingDeque();
 
+
     private GameThread() {
     }
 
+    public static Scheduler io() {
+        return AccessController.doPrivileged((PrivilegedAction<Scheduler>) Schedulers::io);
+    }
+
+    public static Scheduler newThread() {
+        return AccessController.doPrivileged((PrivilegedAction<Scheduler>) Schedulers::newThread);
+    }
+
+    public static Scheduler computation() {
+        return AccessController.doPrivileged((PrivilegedAction<Scheduler>) Schedulers::computation);
+    }
+
```

Comment:
we need to have these in a privileged scope.

Reply:
Why do they need to be in privileged scope? Let's document this here for the future.

Reply:
a lot of thread actions are guarded by the sandbox.